inc/class-wpseo-replace-vars.php

```diff
@@ -1108,6 +1108,76 @@ private function retrieve_userid() {
 		return $replacement;
 	}
 
+	/**
+	 * Retrieve the post/page/cpt's published year for use as replacement string.
+	 *
+	 * @return string
+	 */
+	private function retrieve_post_year() {
+		if ( empty( $this->args->ID ) ) {
+			return null;
+		}
+
+		return \get_the_date( 'Y', $this->args->ID );
+	}
+
+	/**
+	 * Retrieve the post/page/cpt's published month for use as replacement string.
+	 *
+	 * @return string
+	 */
+	private function retrieve_post_month() {
+		if ( empty( $this->args->ID ) ) {
+			return null;
+		}
+
+		return \get_the_date( 'F', $this->args->ID );
+	}
+
+	/**
+	 * Retrieve the post/page/cpt's published day for use as replacement string.
+	 *
+	 * @return string
+	 */
+	private function retrieve_post_day() {
+		if ( empty( $this->args->ID ) ) {
+			return null;
+		}
+
+		return \get_the_date( 'd', $this->args->ID );
+	}
+
+	/**
+	 * Retrieve the post/page/cpt author's first name for use as replacement string.
+	 *
+	 * @return string
+	 */
+	private function retrieve_author_first_name() {
+		return \get_the_author_meta( 'first_name', $this->retrieve_userid() );
```

Comment:
I believe we should add a fallback if `get_the_author_meta`returns an empty string, much like how we do in `retrieve_name` and `retrieve_user_description`.

Also fallback in `retrieve_author_last_name` too

Reply:
Yeah, I did notice that when no first or last name has been set, the output in the editor will be `%%author_first_name%%` or `%%author_last_name%%`. 

FYI, AiOSEO falls back to outputting an empty string in that case (not saying that's what we want necessarily, but something to consider).